tests/pytests/test_client_server_mtls.py

```diff
@@ -0,0 +1,145 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ##############################################################################
+"""
+Basic pytests to exercise stripped-down Client-Server SSL Socket
+communication using mTLS.
+
+# pylint: disable=line-too-long
+
+Sample test program built based on these references:
+
+https://stackoverflow.com/questions/44343230/mutual-ssl-authentication-in-simple-echo-client-server-python-sockets-ssl-m
+https://discuss.python.org/t/ssl-certificate-verify-failed-certificate-verify-failed-ip-address-mismatch-certificate-is-not-valid-for-xxx-xxx-x-xx-ssl-c-997/28403/4
+
+leads to: https://stackoverflow.com/questions/52855924/problems-using-paho-mqtt-client-with-python-3-7
+  where the suggestion to use --addext 'subjectAltName=IP:127.0.0.1' has been provided.
+
+So, finally, generate client / server certificates and private-keys using these commands:
+$ openssl req -new -x509 -days 365 -noenc -out client.pem -keyout client.key --addext 'subjectAltName=IP:127.0.0.1'
+$ openssl req -new -x509 -days 365 -nodes -out server.pem -keyout server.key --addext 'subjectAltName=IP:127.0.0.1'
+# pylint: enable=line-too-long
+"""
+import os
+import socket
+import ssl
+import pprint
+
+# Resolves to current tests/pytests dir
+THIS_SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+
+CERT_HOST = '127.0.0.1'
+CERT_PORT = 1234
+
+# These certificate *.pem and private-key *.key files are generated by
+# the openssl utility.
+SERVER_CERT = THIS_SCRIPT_DIR + '/data/server.pem'
+SERVER_KEYF = THIS_SCRIPT_DIR + '/data/server.key'
+CLIENT_CERT = THIS_SCRIPT_DIR + '/data/client.pem'
+CLIENT_KEYF = THIS_SCRIPT_DIR + '/data/client.key'
+
+# The openssl utility is invoked to specify few fields, where the
+# 'commonName' field is the 3rd field specified; Hence, offset=2
+# This hard-coded field-number arises from the way the openssl req
+# command is run in gen_client_server_certs_key_files.sh
+COMMON_NAME_FIELD = 2
+
+# ##############################################################################
+# To see output, run: pytest --capture=tee-sys -v
+def test_server_process_with_mtls():
+    """
+    Run server-process that listens on an end-point on a secure SSL channel.
+    """
+    context = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+    context.verify_mode = ssl.CERT_REQUIRED
+    context.load_cert_chain(certfile=SERVER_CERT, keyfile=SERVER_KEYF)
+    context.load_verify_locations(cafile=CLIENT_CERT)
```

Comment:
we should use the `POLICY_CERT` for verification; this means that we accept any client which provides a client certificate signed by the Certifier Service